Automattic/jetpack
# Jetpack Social: Prevent SIG from enabling itself automatically on existing posts

## Proposed changes:

This PR fixes an issue where Social Image Generator was being enabled automatically for existing posts, by relying on Gutenberg to update the post meta rather than doing it a hook after saving the post. I've explained the changes in detail in the code since I think the context is helpful.

### Other information:


## Jetpack product discussion


1202816441505085-as-1204236160532606/f

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Make sure Social Image Generator is enabled for your site.
* Create a new post and make sure the Social Image Generator toggle is enabled.
* Publish it or save it as a draft.
* Check that the meta was saved correctly: `get_post_meta( POST_ID )` in the shell
* Now, delete the post meta (`delete_post_meta( POST_ID, '_wpas_options' )` and open the editor for the post.
* Check that the toggle is disabled.
* Change the post title or content and save it.
* Check the meta again, and make sure that SIG is now disabled.
* Repeat this process for the Classic Editor. In this case, SIG should always be disabled.
